### ISSUE TITLE
--auto-correct on windows changes the line-ending

### ISSUE THREAD

**Issue by ccoenen** [bbatsov/rubocop#1566]
I am using Unix-Line-Endings everywhere, but `--auto-correct` writes back CRLF to my files on windows. As far as i can see, there is no setting to change that.

If you are opening a file for writing on windows with filemode "W" (`File.open(name, 'w')`), the resulting file _will_ always have `CRLF` as newline characters. To write `\n`, you would have to open a file for binary writing (`'wb'`)

I found two places where a file was opened for writing, i am not sure which is the right place, but maybe it helps: [formatter_set.rb#add_formatter](https://github.com/bbatsov/rubocop/blob/a28e34e94367ad9dd8b87a1d8c00f60c515dbd6b/lib/rubocop/formatter/formatter_set.rb#L45) and [team.rb#autocorrect](https://github.com/bbatsov/rubocop/blob/a28e34e94367ad9dd8b87a1d8c00f60c515dbd6b/lib/rubocop/cop/team.rb#L70).

As an experiment, i changed this to be 'wb' on my machine, this did the trick. I was rewarded with unix-style newlines.

**Comment by jonas054** [comment]
On my system (Ubuntu Linux), everything still works if I change to `'wb'`. I think it makes most sense to only do this change in `Team#autocorrect`, thus keeping whatever line endings were used. In `FormatterSet#add_formatter` we create a new file, which should have the system default line endings.

**Comment by alexdowad** [comment]
> I was rewarded with unix-style newlines.

Hmm. You could have opened a PR...

**Comment by ccoenen** [comment]
Yes, I could have, but since I had no way of verifying other platforms at that time and my sketchy knowledge of the codebase, i chose not to.

I am glad that you are taking this up again, I have to admit that I did not use `--auto-correct` since, and completely forgot about this issue.

**Comment by erys** [comment]
This still seems to be an issue in current rubocop. Here's my monkey patch solution that seems to work for anyone else having this issue:
```ruby
# lib/rubocop_autocorrect.rb

RuboCop::Cop::Team.class_eval do
  def autocorrect(processed_source, report, original:, offset:)
    @updated_source_file = false
    return unless autocorrect?
    return if report.processed_source.parser_error

    new_source = autocorrect_report(report, original:, offset:)

    return unless new_source

    if @options[:stdin]
      # holds source read in from stdin, when --stdin option is used
      @options[:stdin] = new_source
    else
      filename = processed_source.file_path
      File.write(filename, new_source, mode: 'wb')
    end
    @updated_source_file = true
  end
end
```
and
```yaml
# .rubocop.yml

require:
  - ./lib/rubocop_autocorrect.rb
```
